components/ntp_background_images/browser/ntp_background_images_service.cc

```diff
@@ -169,14 +178,26 @@ void NTPBackgroundImagesService::RegisterBackgroundImagesComponent() {
 }
 
 void NTPBackgroundImagesService::RegisterSponsoredImagesComponent() {
-  const auto data = GetSponsoredImagesComponentData(
-      brave_l10n::GetDefaultISOCountryCodeString());
+  const std::string country_code =
+      brave_l10n::GetCurrentSubdivisionCountryCode(local_pref_);
+
+  const auto data = GetSponsoredImagesComponentData(country_code);
   if (!data) {
     DVLOG(2) << __func__ << ": Not support NTP SI component for "
-             << brave_l10n::GetDefaultLocaleString();
+             << country_code;
+    return;
+  }
+
+  if (sponsored_images_component_id_ == data->component_id.data()) {
```

Comment:
Q) `OnGeoSubdivisionPrefChanged()` could be called when current subdivision country code is not changed?

Reply:
one more Q) If user doesn't change os locale setting, `RegisterSponsoredImagesComponent()` is called only once during the runtime?

Reply:
`OnGeoSubdivisionPrefChanged` is called only when the `brave.geo.subdivision` local state pref is changed.

If `brave.geo.subdivision` is not changing, then RegisterSponsoredImagesComponent() will be called only once, during browser startup.

`brave.geo.subdivision` is set based on the response we got from https://geo.ads.brave.com/v1/getstate. The typical reponse is `{"country":"US","region":"FL"}`

Reply:
I think it could be called twice when `kGeoSubdivision` is empty string at startup.
First call with `GetDefaultISOCountryCodeString()` as `kGeoSubdivisoin` is empty.
and then second call after `kGeoSubdivision` is set if it's different with default one.
Then, we need to download sponsored images component twice after startup.
Curious my assumption is available scenario.

If so, how about calling `RegisterBackgroundImagesComponent();` in ctor when `kGeoSubdivision` is not empty?

Reply:
Yes, this is a possible scenario. I will do a fix based on your suggestion.

Reply:
@simonhong, after discussion with @tmancey we found that this scenario is not possible because `kGeoRegionCode` pref is only applied when the user joins Brave Rewards. This is not a fresh profile and the resource for the system locale is already downloaded.
If user is not joined Brave Rewards, then the value of `kGeoRegionCode` pref is equal to `GetDefaultISOCountryCodeString()`. So we have the same behavior as before.

Reply:
Thanks for explaination. My below understanding is right? :) 
User will use default region code (`GetDefaultISOCountryCodeString()`) from the start.
and it could be changed when rewards is enabled if https://geo.ads.brave.com/v1/getstate gives different region. 
Then, user could get another NTP components for new region code.
If user disable rewards, user will get default region's NTP components again.

Reply:
Yes, that's correct. I've updated the test plan with the reset step